libavcodec/libaribcaption.c

```diff
@@ -0,0 +1,1141 @@
+             "Format: Name, "
+             "Fontname, Fontsize, "
+             "PrimaryColour, SecondaryColour, OutlineColour, BackColour, "
+             "Bold, Italic, Underline, StrikeOut, "
+             "ScaleX, ScaleY, "
+             "Spacing, Angle, "
+             "BorderStyle, Outline, Shadow, "
+             "Alignment, MarginL, MarginR, MarginV, "
+             "Encoding\r\n"
+
+             "Style: "
+             "Default,"             /* Name */
+             "%s,%d,"               /* Font{name,size} */
+             "&H%x,&H%x,&H%x,&H%x," /* {Primary,Secondary,Outline,Back}Colour */
+             "%d,%d,%d,0,"          /* Bold, Italic, Underline, StrikeOut */
+             "100,100,"             /* Scale{X,Y} */
+             "0,0,"                 /* Spacing, Angle */
+             "%d,%d,%d,"            /* BorderStyle, Outline, Shadow */
+             "%d,10,10,10,"         /* Alignment, Margin[LRV] */
+             "0\r\n"                /* Encoding */
+             "\r\n"
+
+             "[Events]\r\n"
+             "Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text\r\n",
+            ctx->plane_width, ctx->plane_height,
+            font_name, ctx->font_size,
+            ASS_DEFAULT_COLOR, ASS_DEFAULT_COLOR,
+            ASS_DEFAULT_BACK_COLOR, ASS_DEFAULT_BACK_COLOR,
+            -ASS_DEFAULT_BOLD, -ASS_DEFAULT_ITALIC, -ASS_DEFAULT_UNDERLINE,
+            ctx->border_style, outline, shadow, ASS_DEFAULT_ALIGNMENT);
+
+    av_freep(&font_name);
+    if (!avctx->subtitle_header)
+        return AVERROR(ENOMEM);
+    avctx->subtitle_header_size = strlen(avctx->subtitle_header);
+    return 0;
+}
+
+static void set_ass_color(AVBPrint *buf, int color_num,
+                          aribcc_color_t new_color, aribcc_color_t old_color)
+{
+    if (ARIBCC_COLOR_DIFF_RGB(new_color, old_color))
+        av_bprintf(buf, "{\\%dc&H%06x&}", color_num,
+                                          ARIBCC_COLOR_RGB(new_color));
+    if (ARIBCC_COLOR_DIFF_A(new_color, old_color) && ARIBCC_COLOR_A(new_color))
+        av_bprintf(buf, "{\\%da&H%02x&}", color_num,
+                                          ARIBCC_COLOR_A(new_color));
+}
+
+static int aribcaption_trans_ass_subtitle(ARIBCaptionContext *ctx)
+{
+    AVSubtitle *sub = ctx->sub;
+    AVBPrint buf;
+    int ret = 0, rect_idx;
+
+    if (ctx->caption.plane_width > 0 && ctx->caption.plane_height > 0 &&
+        (ctx->caption.plane_width != ctx->plane_width ||
+         ctx->caption.plane_height != ctx->plane_height)) {
+        ctx->plane_width = ctx->caption.plane_width;
+        ctx->plane_height = ctx->caption.plane_height;
+        if ((ret = set_ass_header(ctx)) < 0)
+            return ret;
+    }
+
+    sub->format = 1; /* text */
+    if (ctx->caption.region_count == 0) {
+        /* clear previous caption for indefinite duration  */
+        ff_ass_add_rect(sub, "{\\r}", ctx->readorder++, 0, NULL, NULL);
+        return 1;
```

Comment:
The `{\r}` shouldn't be necessary here; you should be able to just use an empty string.